src/components/TodoModal/TodoModal.tsx

```diff
@@ -1,39 +1,70 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import { Loader } from '../Loader';
+import { getUser } from '../../api';
+import { Todo } from '../../types/Todo';
+import { User } from '../../types/User';
+
+type Props = {
+  selectedTodo: Todo;
+  setSelectedTodo: (todo: Todo | null) => void;
+};
+
+export const TodoModal: React.FC<Props> = ({
+  selectedTodo,
+  setSelectedTodo,
+}) => {
+  const [loader, setLoader] = useState(true);
+  const [user, setUser] = useState<User | null>(null);
```

Comment:
Initializing state with props is not recommended as it can lead to bugs and inconsistencies in the state. Instead, you should use the prop directly or if necessary, synchronize state with props using the `useEffect` hook.